internal/datacoord/index_service.go

```diff
@@ -337,8 +341,9 @@ func (s *Server) completeIndexInfo(indexInfo *indexpb.IndexInfo, index *model.In
 			continue
 		}
 
-		//data before index create time should create complete
-		if seg.GetStartPosition().GetTimestamp() > index.CreateTime {
+		// if realTime, calculate current statistics
+		// if not realTime, skip segments created after index create
+		if seg.GetStartPosition().GetTimestamp() > index.CreateTime && !realTime {
```

Comment:
if  !realTime && seg.GetStartPosition().GetTimestamp() > index.CreateTime

for better readability